services/lumberjack/src/service.js

```diff
@@ -0,0 +1,293 @@
+    try {
+      ping =
+        (await request.get('http://' + this._pingHost + ':' +
+          this._pingPort + '/api/ping')
+          .proto(stats.PingTimes)
+          .timeout(1000)).body;
+    } catch (err) {
+      logger.error(err);
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: 0 },
+          tags: { host: 'non-existent-service', port: 0, name: 'service'}
+        }], {
+        database: this._dbName
+      });
+    }
+
+    // Write data for services
+    for (let endpoint of ping.service_pings) {
+      const { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: this._dbName
+      });
+    }
+
+    // Write data for devices
+    for (let endpoint of ping.device_pings) {
+      const { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { devicePing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: this._dbName
+      });
+    }
+  }
+
+  /** Get telemetry upload rate data and write to the database */
+  async _uploadRate() {
+    // Get upload rate
+    let rate;
+    try {
+      rate =
+      (await request.get('http://' + this._forwardInteropHost + ':' +
+        this._forwardInteropPort + '/api/upload-rate')
+        .proto(stats.InteropUploadRate)
+        .timeout(1000)).body;
+    } catch (err) {
+      await this._influx.writeMeasurement('upload-rate', [
+        {
+          fields: { total_1: 0, total_5: 0, fresh_1: 0, fresh_5: 0},
+          tags: { host: 'non-existent-service', port: 0}
+        }], {
+        database: this._dbName
+      });
+    }
+
+    await this._influx.writeMeasurement('upload-rate', [
+      {
+        fields: { total_1: rate.total_1, total_5: rate.total_5,
+          fresh_1: rate.fresh_1, fresh_5: rate.fresh_5 },
+        tags: { host: this._forwardInteropHost, port: this._forwardInteropPort }
+      }], {
+      database: this._dbName
+    });
+  }
+
+  /** Get ground telemetry overview and write to the database */
+  async _groundTelemetry() {
+    const OFFLINE = 0;
+    const ONLINE = 1;
+    let gstatus;
+
+    // Get telemetry overview
+    let groundTelem;
+    try {
+      groundTelem =
+        (await request.get('http://' + this._groundTelemetryHost + ':' +
+          this._groundTelemetryPort + '/api/overview')
+          .proto(telemetry.Overview)
+          .timeout(1000)).body;
+    } catch (err) {
+      gstatus = OFFLINE;
+    }
+
+    // Check if current time is the same or less than the previous
+    // timestate
+    // Update time if current time is greater than previous timestate
+    // this._gTimes will be set to groundTelem.time at start since
+    // if/else will evaluate to false when this._gTimes is undefined
+    if (groundTelem.time <= this._gTimes) {
```

Comment:
If the service is offline this will raise an exception because `groundTelem` is `undefined`.